### DISCUSSION TITLE
How can I use flat config to only apply eslint:recommended to specific files?

### DISCUSSION THREAD

**Discussion by lixiaoyan** [eslint/eslint#16359] [Q&A]
Is there an equivalent to this?

```
{
  "overrides": [
    {
      "files": ["foo.js"],
      "extends": ["eslint:recommended"]
    }
  ]
}
```

**Comment by AndreaPontrandolfo** [comment]
just add a new object and disable the rules you aren't interested in.

**Comment by mdjermanovic** [comment]
I don't think there's an equivalent to this in the flat config. The only comment I could find in the RFC that may be related to this problem is https://github.com/eslint/rfcs/pull/9#issuecomment-478293225, but that part doesn't seem to be addressed.

@nzakas is this use case intentionally not supported in the flat config?

**Reply by nzakas** [reply]
Correct, we don’t have an exact analogy for this use case right now.

One approach I’ve been thinking about is to export a plugin-like object from ESLint that lets you directly access internal rules and configs just like you would with a plugin. So something like:

```js
const eslintPlugin = require(“eslint/plugin”);

module.exports = {
    files: [“foo.js”],
    rules: {
        …eslintPlugin.configs.recommended.rules,
        semi: [“warn”, “never”]
    }
}
```

Need to think on it a bit more though.

**Reply by mdjermanovic** [reply]
Another approach could be to add built-in configs to the `context` argument of [function configs](https://github.com/eslint/rfcs/blob/893b49194689fcf0d34743eac75773d098264656/designs/2019-config-simplification/README.md?plain=1#L607) (if and when we implement that feature):

```js
module.exports = context => [{
    files: ["foo.js"],
    rules: {
        ...context.configs.recommended.rules
    }
}];
```

or with [including function configs in an array](https://github.com/eslint/rfcs/blob/893b49194689fcf0d34743eac75773d098264656/designs/2019-config-simplification/README.md?plain=1#L636):
 
```js
module.exports = [
    context => ({
        files: ["foo.js"],
        rules: {
            ...context.configs.recommended.rules
        }
    })    
];
```

**Reply by nzakas** [reply]
Yeah, I'm still ambivalent about introducing function configs. To the extent that we are encouraging plugins to be used in a particular way, it seems like less cognitive overhead to create a similar way to work with builtin ESLint functionality that maps well to plugins (rules, configs).

**Comment by nzakas** [comment]
Related: https://github.com/eslint/eslint/issues/16537
